lib/rules/best-practises/no-global-import.js

```diff
@@ -0,0 +1,43 @@
+const BaseChecker = require('../base-checker')
+
+const ruleId = 'no-global-import'
+const meta = {
+  type: 'best-practises',
+
+  docs: {
+    description: 'Import statement includes an entire file instead of selected symbols',
+    category: 'Best Practise Rules',
+    examples: {
+      bad: [
+        { description: 'import all members from a file', code: 'import * from "foo.sol"' },
+        { description: 'import an entire file', code: 'import "foo.sol"' },
+      ],
+      good: [{ description: 'import names explicitly', code: 'import {A} from "./A.sol"' }],
+    },
+  },
+
+  isDefault: false,
+  recommended: true,
+  defaultSetup: 'warn',
+
+  schema: null,
+}
+
+class NoGlobalImportsChecker extends BaseChecker {
+  constructor(reporter) {
+    super(reporter, ruleId, meta)
+  }
+
+  SourceUnit(node) {
+    node.children.forEach((it) => {
+      if (it.type === 'ImportDirective' && !it.symbolAliases) {
```

Comment:
It would be better to use the `ImportDirective` visitor callback instead of iterating here. Not a big difference, but it's more idiomatic.

Reply:
Initially I had an issue by which the ImportDirective callback wasn't being called for the nodes not defining  `symbolAliases`, which are the interesting ones. But I can't replicate it now :upside_down_face:  so it was probably something else.

pushed again using the ImportDirective callback as suggested.

Reply:
seems to me this can be resolved and pr approved 
do you agree @fvictorio ?